sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/TypedSchemaTransformProvider.java

```diff
@@ -61,7 +63,13 @@ Optional<List<String>> dependencies(ConfigT configuration, PipelineOptions optio
   @Override
   public final Schema configurationSchema() {
     try {
-      return SchemaRegistry.createDefault().getSchema(configurationClass());
+      Schema schema = SchemaRegistry.createDefault().getSchema(configurationClass());
+
+      // Sort the fields by name to ensure a consistent schema is produced
+      Schema sortedSchema = schema.getFields().stream()
```

Comment:
We should also preserve other fields of the Schema, for example, id, options, encoding_positions_set (and any future fields).

https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/schema.proto#L40